### ISSUE TITLE
sails.config.paths

### ISSUE THREAD

**Issue by maheshchari** [balderdashy/sails#2965]
I think it better multiple asset folders should be configured for www hook . 
sails.config.paths = [path1,path2]

**Comment by Dhumez-Sebastien** [comment]
:+1:

**Comment by maheshchari** [comment]
and it good to have any configuration option can be a "String" or "Array" like grails configuration.

**Comment by maheshchari** [comment]
I have tried to override default www middle ware but failed :( 
Is there any example to override middle ware or hooks in sails.

**Comment by Dhumez-Sebastien** [comment]
Ok i've found a solution : 

#### Add staticRoutes into config

Here you can set a list of static folder : 

```javascript
// config/staticRoutes.js

var path = require('path');

module.exports.staticRoutes = {
    staticFolder1: function(app){
        console.log('Load custom folder 1');
        app.use('/var', require('../node_modules/sails/node_modules/express')['static']( path.normalize(__dirname + '/../var') ) );
    },
    staticFolder2: function(app){
        console.log('Load custom folder 2');
        app.use('/var2', require('../node_modules/sails/node_modules/express')['static']( path.normalize(__dirname + '/../var2') ) );
    },
    staticFolder3: function(app){
        console.log('Load custom folder 3');
        app.use('/var3', require('../node_modules/sails/node_modules/express')['static']( path.normalize(__dirname + '/../var3') ) );
    }
};
```

#### Load all routes into custom middleware

```javascript
// config/http.js

    customMiddleware: function(app){
        _.each(sails.config.staticRoutes, function (pathKey) {
            pathKey(app);
        });
    },

    middleware: {

        passportInit    : require('passport').initialize(),
        passportSession : require('passport').session(),

        order: [
            'startRequestTimer',
            'cookieParser',
            'session',
            'passportInit',
            'passportSession',
            'myRequestLogger',
            'bodyParser',
            'handleBodyParserError',
            'compress',
            'methodOverride',
            'poweredBy',
            'router',
            '$custom',    // <----- DON'T FORGET THAT
            'www',
            'favicon',
            '404',
            '500'
        ]
    }
```

**Comment by maheshchari** [comment]
Thanks @Dhumez-Sebastien

**Comment by maheshchari** [comment]
Can we override www middle ware?

**Comment by sailsbot** [comment]
Thanks for posting, @maheshchari.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by Wizzercn** [comment]
@Dhumez-Sebastien  help...

app.use is undefined 

error: Sending 500 ("Server Error") response: 
 TypeError: undefined is not a function
    at module.exports.staticRoutes.staticFolder1 (/node/nodeshop/config/staticRoutes.js:9:9)

**Comment by RWOverdijk** [comment]
@Wizzercn You probably didn't follow all instructions :) App is being passed to customMiddleware. Make sure you're using it and that you're allowing parameters to be sent.
